Conditional restart

flichtenheld opened the pull request:
<!--
Thank you for contributing to this project!

- This project has a Contributor Code of Conduct: https://voxpupuli.org/coc/
- Please check that here is no existing issue or PR that addresses your problem.
- Our vulnerabilities reporting process is at https://voxpupuli.org/security/

-->
#### Pull Request (PR) description

Rebase on latest upstream.

#### This Pull Request (PR) fixes the following issues

Debian GPG key outdated.

ekohl committed:
Rewrite function tests to modern rspec-puppet

ekohl committed:
Clean up acceptance spec helper

ekohl committed:
Simplify the fact definition

This removes the entire module for the facts. The benefit is that it's
much easier to see what's going on. There is still testing which means
there's no reduction in code coverage.

mattock committed:
Allow disabling automatic Jenkins master restarts even if service is managed

This is useful on production Jenkins masters where non-scheduled restarts are
dangerous. Unit file changes and user changes that happen before tha package is
installed are not affected.

ekohl committed:
Remove duplicate resources in acceptance tests

These plugins are also in the default plugins so don't need to be
defined again.

ekohl committed:
Merge pull request #947 from ekohl/simplify-fact

Simplify the fact definition

ekohl committed:
Merge pull request #931 from ekohl/spec-helper-acceptance

Clean up acceptance spec helper

ekohl committed:
Fix syntax error in spec example

bastelfreak committed:
modulesync 4.0.0

ekohl committed:
Use voxpupuli-acceptance

TomRitserveldt committed:
add new expected cli auth error

TomRitserveldt committed:
Add test for new cli auth error as well

ekohl committed:
Merge pull request #965 from vrtdev/feature/cli-auth-fix

Feature/cli auth fix

anesterova committed:
Use the correct Jenkins CLI location

This is needed with recent Jenkins versions and compatible with old
versions, as least dating back to 2.99

Fixes #967
Fixes #944

ekohl committed:
Merge pull request #958 from ekohl/disable-plugin-acceptance-tests

Fix broken acceptance tests

igalic committed:
Update debian GPG key

This fixes #971.

ekohl committed:
Merge pull request #972 from voxpupuli/fix/debian-gpg-key

Update debian GPG key

ekohl committed:
Merge pull request #963 from voxpupuli/rewrite-acceptance-tests

Use voxpupuli-acceptance

ekohl committed:
Remove old nodesets

ekohl committed:
Use voxpupuli-test ~> 1.4 for now

ekohl committed:
Fix BasicSSHUserPrivateKey acceptance test

ekohl committed:
Fix systemd handling

03ad333690e7925240d2c48111f700d36beebbc6 removed the static setting of
systemd facts in tests. This fixes that and some other variable handling
around systemd.

ekohl committed:
Disable RSpec/MultipleExpectations

This follows the modulesync repo.

ekohl committed:
Disable Rspec/MessageSpies

For now it's too much work to convert these.

ekohl committed:
Merge pull request #978 from ekohl/fix-test

Fix BasicSSHUserPrivateKey acceptance test

ekohl committed:
Use puppet-strings formatting in jenkins::plugin

ekohl committed:
Add an extension parameter to jenkins::plugin

Not all plugins are .hpi. This allows managing plugins with .jpi, like
jdk-tool.

ekohl committed:
Rewrite the plugin acceptance tests to jquery3-api

The jquery3-api plugin has no dependencies, which makes it much easier
to work with.

The explicitly managed jdk-tool and trilead-api should probably be
managed at the top level as default plugins, but for some reason they
can't be properly managed. This at least makes the tests work again.

ekohl committed:
Remove deprecated jenkins::plugins parameters

ekohl committed:
Remove $manage_config parameter

The same can be achieved by only operating if config_filename is set.

ekohl committed:
Document all jenkins::plugin parameters

ekohl committed:
Make jenkins::plugin puppet-lint clean

ekohl committed:
Merge pull request #981 from ekohl/fix-plugin-tests

Fix the plugin acceptance tests

ekohl committed:
Merge pull request #982 from ekohl/function-tests

Rewrite function tests to modern rspec-puppet

ekohl committed:
Drop code old CLI remoting support

Jenkins 2.54 was released in 2017, which is long EOL. Current versions
have even dropped support for the old remoting.

ekohl committed:
Merge pull request #984 from ekohl/drop-old-remoting

Drop code old CLI remoting support

ekohl committed:
Remove use of validate_bool in example

ekohl committed:
Merge pull request #985 from ekohl/jenkins-plugin-cleanup

Remove params from jenkins::plugin and make it lint clean

ekohl committed:
Implement ConduitCredentialsImpl in groovy_helper

ekohl committed:
Merge pull request #986 from ekohl/ConduitCredentialsImpl

Implement ConduitCredentialsImpl in groovy_helper

ekohl committed:
Drop GoogleRobotPrivateKeyCredentials implementation

This implementation is broken and as preparation for 3.0.0 the code base
is being reduced to working versions. If this is desired by users, a
working patch should be provided.

ekohl committed:
Merge pull request #987 from ekohl/fix-google-impl

Drop GoogleRobotPrivateKeyCredentials implementation

ekohl committed:
Remove deprecated enabled parameter on job

9ef227e97806ba387ba2b9ea82e0ac543376415a deprecated this and made it a
noop. This removes the support.

saz committed:
allow puppetlabs/java < 7.0.0

ekohl committed:
Lint clean and add more puppet-strings docs

ekohl committed:
Merge pull request #995 from ekohl/lint

Lint clean and add more puppet-strings docs

ekohl committed:
Merge pull request #994 from RegioHelden/master

allow puppetlabs/java < 7.0.0

ekohl committed:
puppet-lint autofix

ekohl committed:
Update Ubuntu and EL support to supported versions

bastelfreak committed:
centos: add pidfile workaround

ekohl committed:
Merge pull request #962 from voxpupuli/modulesync

modulesync 4.0.0

genebean committed:
Change supported Puppet versions to 6 & 7

ekohl committed:
Merge pull request #991 from ekohl/remove-enabled-parameter

Remove deprecated enabled parameter on job

ekohl committed:
Merge pull request #1003 from genebean/puppet_versions

Change supported Puppet versions to 6 & 7

saz committed:
update dependency puppetlabs-java to support < 8.0.0

ekohl committed:
Merge pull request #1008 from RegioHelden/java_dep_version

update dependency puppetlabs-java to support < 8.0.0

bastelfreak committed:
puppet/archive: allow 5.x

vStone committed:
Fix slave run with additional swarm arguments provided

Without this, all arguments are added in a single string.

ekohl committed:
Merge pull request #1013 from UGent-DICT/bugfix/swarm_args

Fix slave run with additional swarm arguments provided

bastelfreak committed:
puppetlabs/stdlib: allow 7.x

bastelfreak committed:
puppetlabs/apt: allow 8.x

bastelfreak committed:
puppet/zypprepo: Allow 3.x & 4.x

bastelfreak committed:
camptocamp/systemd: Allow 3.x

bastelfreak committed:
switch from camptocamp/systemd to voxpupuli/systemd

smortex committed:
Allow stdlib 8.0.0

smortex committed:
Allow archive 6.x

bastelfreak committed:
puppet-lint: fix top_scope_facts warnings

bastelfreak committed:
Merge pull request #1020 from bastelfreak/puppet-lint-top_scope_facts-check

puppet-lint: fix top_scope_facts warnings

bastelfreak committed:
Merge pull request #1018 from voxpupuli/systemd_voxpupuli

switch from camptocamp/systemd to voxpupuli/systemd

bastelfreak committed:
Merge branch 'master' into systemd

bastelfreak committed:
Merge pull request #1017 from bastelfreak/systemd

camptocamp/systemd: Allow 3.x

fe80 committed:
add version fact and systemd_type param

ekohl committed:
Drop EL6 tests

In 5bd3da4fc990095ab6cc244a241e7761fa8143f4 support for EL6 was dropped,
but the tests weren't dropped.

Fixes: 5bd3da4fc990095ab6cc244a241e7761fa8143f4

ekohl committed:
Remove Optional[] where the param has a default

This is found by the optional_default check.

ekohl committed:
Rewrite on_supported_os checks

Previously it would run on all operating systems where FacterDB had an
entry. This also included unsupported versions such as Ubuntu 21.10.

ekohl committed:
Merge pull request #1027 from ekohl/fix-lint-optional-default

Remove Optional[] where the param has a default

ekohl committed:
Merge pull request #1026 from ekohl/drop-el6-tests

Drop EL6 tests

fe80 committed:
Merge remote-tracking branch 'upstream/master' into add_daemon_mode

ekohl committed:
Merge pull request #1025 from fe80/add_daemon_mode

add version fact and daemon param for systemd unit file

jvdmr committed:
Don't put unparseable JSON in output

Don't put unparseable JSON in output as it may still contain valid credentials
or other sensitive data that should not be added to logs in plaintext.

ekohl committed:
Move static params to the top level

These values are static and moving them into the file makes things more
obvious. First of all, puppet-strings can now show the defaults. As a
developer it's also easier to trace things back.

ekohl committed:
Fix unit tests with kwarg expectations

Ruby 3.0 changes behavior around keyword arguments[1]. This somehow
broke expectations in the test suite and possibly at runtime with Ruby 3
but that's hard to verify.

There is an ugly Ruby < 2.7 compatibility hack that I didn't know how
to solve in a cleaner way.

[1]: https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

bastelfreak committed:
Merge pull request #1034 from ekohl/kwargs

Fix unit tests with kwarg expectations

ekohl committed:
Drop sysvinit support in jenkins::slave

For the OS families where a template was provided now default to using
systemd and we don't test with sysvinit.

This drops the ability to smoothly transition from sysvinit to systemd,
but that's not really important.  Very few users will switch their
system from sysvinit to systemd and if they do, it's a bigger upgrade
anyway.

It doesn't implement any cleanup since it's not expected to be present.

ekohl committed:
Move to native systemd integration

Jenkins 2.332 started to include a systemd unit file which means we can
greatly simplify this module. It aligns all distros on a single
standard.

No attempt is made at supporting Jenkins < 2.332 but the old override
file is cleaned up.

ekohl committed:
More puppet-strings conversion

ekohl committed:
Use hpi for jdk-tool and trilead-api plugins

The latest versions are publishedd as hpi, which is the default for
jenkins::plugin.

ekohl committed:
Add missing plugin dependencies to default_plugins

Without these plugins the plugin purging removes too much.

ekohl committed:
Merge pull request #1036 from ekohl/puppet-strings-tag

More puppet-strings conversion

ekohl committed:
Correct documentation of jenkins::cli

I can't quite make out how this was ever true. Looking at
cfb33eff25f3275eaed65173448215c55b0fd000 the location was always
$libdir/jenkins-cli.jar.

Since 5ef894f60b2855328410b00771fd2c4779dc69f8 unzip is no longer
needed.

mattock commented:
I'm all for rebasing with upstream. @kibahop ok for you? You've used this module the most...

ekohl committed:
Merge pull request #1037 from ekohl/move-to-top-level

Move static params to the top level

ekohl committed:
Set the correct SELinux type for jenkins-slave-run

Without this systemd refuses to start.

ekohl committed:
Drop Ubuntu 16.04 support

ekohl committed:
Install lsb-release on acceptance tests if needed

ekohl committed:
Merge pull request #1035 from ekohl/native-systemd

Rewrite to native systemd

ekohl committed:
Modulesync 5.2.0

ekohl committed:
puppet-lint fixes

ekohl committed:
Run rubocop --auto-correct

ekohl committed:
Remove redundant service verification

Fixes: cb71c5d3561b46339d58b0c79082a8a762cfeed8

ekohl committed:
Manual Rubocop compliance fixes

ekohl committed:
Manage Rubocop config outside of modulesync

This module doesn't comply with out Rubocop config, but fixing it takes
quite a bit of effort. For now we can disable the cops to at least allow
using most of our config.

ekohl committed:
Run rubocop --auto-correct-all

ekohl committed:
Remove shared context jenkins

After 98f46ca999635d0a9a2f4ad84d70ed60389f7b47 it was largely redundant
anyway. It also changes the case-when with a hash as suggested by
Rubocop.

ekohl committed:
Remove duplicate test

ekohl committed:
Run acceptance tests with weekly releases

The test suite is broken with LTS due to the credentials plugin. This
should be reverted once LTS is new enough to work again.

ekohl committed:
Correct plugin dependencies

The plugins have new dependencies. This completes them to allow the
tests to pass. There is one pending issue because the credentials really
aren't idempotent.

ekohl committed:
Correct setup acceptance helper filename

Fixes: 860af1e2e85f104904ed0e393387a6e38f254292

ekohl committed:
Merge pull request #1040 from voxpupuli/modulesync

Modulesync 5.2.0

ekohl committed:
Rewrite skipped test to a correct syntax

This test was using an invalid syntax. To test if it was still failing I
executed it, but found it couldn't work. The update-job is still doesn't
implement this. The if: false hack is used to skip since nested contexts
are used and you can't easily skip those otherwise.

ekohl committed:
Move systemd overrides to user_setup for Debian

On Debian services are started automatically during package
installation. The setup wizard is configured via the systemd override so
creating this prior to package installation works around it. On
non-Debian (where it isn't autostarted) it doesn't matter and managing
all platforms the same has benefits.

The notify_service was missing which caused the process to run with an
incorrect parameter. This also adds a test to verify the argument is
really there when running. It should be noted that on Ubuntu 20.04 I
noticed a weird bug where it wasn't picked up. The test did catch it so
when the OS version support is added it'll be easier to catch.

Fixes: 98f46ca999635d0a9a2f4ad84d70ed60389f7b47

ekohl committed:
Remove showing of jenkins_plugins fact

facter doesn't load the Puppet modules so this fact is never set.

ekohl committed:
Replace apply2 usage with a shared example

voxpupuli-acceptance provides the 'an idempotent resource' shared
example which is effectively the same, just more standard.

jan-win1993 committed:
Drop-in file systemd configuration

ekohl committed:
Merge pull request #1041 from ekohl/fix-acceptance-tests

Fix acceptance tests

ekohl committed:
Merge pull request #1019 from voxpupuli/stdlib

Allow up-to-date dependencies

ekohl committed:
Merge pull request #1044 from jan-win1993/systemd-dropin-config

Drop-in file systemd configuration

flichtenheld committed:
Update Debian repository key for 2023

Signed-off-by: Frank Lichtenheld <frank@lichtenheld.com>
